src/target/riscv/riscv-013.c

```diff
@@ -896,6 +908,10 @@ static int wait_for_idle(struct target *target, uint32_t *abstractcs)
 		riscv_command_timeout_sec,
 		*abstractcs);
 
+	if (!dm->abtractcs_busy)
+		LOG_TARGET_ERROR(target, "BUG: dm->abtractcs_busy was not set in time!");
+	dm->abtractcs_busy = true;
+
```

Comment:
I would like to leave it as-is.

It is convenient to use `wait_for_idle()` as a default error handler in case there was a call to a subroutine which may execute an abstract command but may fail before executing the command.

I am not ready to pledge that that there are no such uses currently. 

IMHO, it is desirable for `wait_for_idle()` to be a NOP (apart from one read of `abstractcs`) in case there was no abstract command execution to begin with.

Reply:
Ok, understood, thank you for explanation.

Still, could we still change the wording a bit to make it clearer what has happened? For example as shown below:

```c
	if (!dm->abstract_cmd_maybe_busy)
		LOG_TARGET_ERROR(target, "BUG: dm->abstract_cmd_maybe_busy had not been set when starting an abstract command");
	dm->abstract_cmd_maybe_busy = true;
```

Reply:
Addressed.